src/js/client/editprefset-productList-widget.js

```diff
@@ -0,0 +1,149 @@
+/*
+ * editprefset-productList-widget
+ *
+ * A component that renders a list of all the applications/products, as well
+ * as including Generic Settings at the top of the list. Typically used on
+ * the left hand side for navigation and adding new products to an NP Set.
+ */
+/* global lunr */
+"use strict";
+
+var gpii  = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.devpmt");
+fluid.registerNamespace("gpii.devpmt.productList");
+
+/**
+ *  Main infusion component for a filterable product list.
+ */
+fluid.defaults("gpii.devpmt.productListWidget", {
+    gradeNames: ["gpii.handlebars.templateAware", "gpii.binder.bindOnCreate", "gpii.binder.bindOnDomChange"],
+    model: {
+        productListFilter: "",
+        allSolutionsSorted: []
+    },
+    modelListeners: {
+        productListFilter: {
+            func: "{that}.filterProductList"
+        },
+        allSolutionsSorted: {
+            func: "{that}.updateLunrIndex"
+        }
+    },
+    selectors: {
+        productListFilterContainer: "#filter-container",
+        productListItems: ".pmt-sidebar-toc-product",
+        productListLinks: ".pmt-sidebar-product-link"
+    },
+    bindings: {
+    },
+    templates: {
+        initial: "editprefset-productList-widget"
+    },
+    components: {
+        filter: {
+            type: "gpii.devpmt.filterWidget",
+            createOnEvent: "onMarkupRendered",
+            container: "{that}.dom.productListFilterContainer",
+            options: {
+                selectors: {
+                    initial: "#common-filter-area"
+                },
+                model: {
+                    filterText: "{productListWidget}.model.productListFilter"
+                }
+            }
+        }
+    },
+    lunrIndex: null,
+    invokers: {
+        renderInitialMarkup: {
```

Comment:
Looks like the implementation of "renderInitialMarkup" can be factored out across many components

Reply:
This has been factored out into my base `gpii.devpmt.viewComponent`